frontend/src/pages/pipelines/global/experiments/executions/details/ExecutionDetailsInputOutputSection.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { Bullseye, Spinner, Stack, StackItem, Title } from '@patternfly/react-core';
+import { Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import { Event } from '~/third_party/mlmd';
+import { useGetLinkedArtifactsByEvents } from '~/concepts/pipelines/apiHooks/mlmd/useGetLinkedArtifactsByEvents';
+import { getArtifactNameFromEvent } from '~/pages/pipelines/global/experiments/executions/utils';
+
+type ExecutionDetailsInputOutputSectionProps = {
+  isLoaded: boolean;
+  title: string;
+  events: Event[];
+  artifactTypeMap: Record<number, string>;
+};
+
+const ExecutionDetailsInputOutputSection: React.FC<ExecutionDetailsInputOutputSectionProps> = ({
+  isLoaded,
+  title,
+  events,
+  artifactTypeMap,
+}) => {
+  const [linkedArtifacts, isLinkedArtifactsLoaded] = useGetLinkedArtifactsByEvents(events);
+
+  if (!isLoaded || !isLinkedArtifactsLoaded) {
+    return (
+      <Bullseye>
+        <Spinner />
+      </Bullseye>
+    );
+  }
+
+  const artifactDataMap: Record<number, { name: string; typeId: number | undefined; uri: string }> =
+    {};
+  linkedArtifacts.forEach((linkedArtifact) => {
+    const id = linkedArtifact.event.getArtifactId();
+    if (!id) {
+      return;
+    }
+    artifactDataMap[id] = {
+      name: getArtifactNameFromEvent(linkedArtifact.event) || '',
+      typeId: linkedArtifact.artifact.getTypeId(),
+      uri: linkedArtifact.artifact.getUri() || '',
+    };
+  });
+
+  return (
+    <Stack hasGutter>
+      <StackItem>
+        <Title headingLevel="h3">{title}</Title>
+      </StackItem>
+      <StackItem>
+        {events.length === 0 ? (
+          `No ${title.toLowerCase()}`
+        ) : (
+          <Table
+            aria-label={`Execution details ${title.toLowerCase()} table`}
+            variant="compact"
+            borders={false}
+          >
+            <Thead>
+              <Tr>
+                <Th width={10}>Artifact ID</Th>
+                <Th width={15}>Name</Th>
+                <Th width={15}>Type</Th>
+                <Th width={60}>URI</Th>
+              </Tr>
+            </Thead>
+            <Tbody>
+              {events.map((event) => {
+                const id = event.getArtifactId();
+                const data = artifactDataMap[id];
+                const type = data.typeId ? artifactTypeMap[data.typeId] : null;
+                return (
+                  <Tr key={id}>
+                    <Td dataLabel="Artifact ID">{id}</Td>
+                    {/* TODO: add artifact details page line */}
```

Comment:
is this TODO still relevant, if so is it logged

Reply:
It's related to Jeff's PR https://github.com/opendatahub-io/odh-dashboard/pull/2765, once we get it merged in we can add that link. I leave it here because if his PR is merged before mine, I can simply rebase it and update the code.